alice/middleware/logger.go

```diff
@@ -22,35 +27,32 @@ type logger struct {
 	logRequests bool
 }
 
-type loggingResponseWriter struct {
-	http.ResponseWriter
-	StatusCode int
-}
-
-func (l *loggingResponseWriter) WriteHeader(code int) {
-	l.StatusCode = code
-	l.ResponseWriter.WriteHeader(code)
-}
-
 // NewLogrusLogger allows you to setup a base entry to use for logging
 func NewLogrusLogger(baseEntry *logrus.Entry, logRequests bool) Logger {
 	return &logger{entry: baseEntry, logRequests: logRequests}
 }
 
 func (l *logger) Log(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		entry := l.entry.WithFields(logrus.Fields{
-			logFieldRequestID: getRequestIDFromContext(r.Context()),
-			logFieldUserID:    getInsecureUserIDFromContext(r.Context()),
-		})
+		loggingResponseWriter := &lrw.LoggingResponseWriter{w, http.StatusOK, nil}
+
 		// add logger to the context
 		ctx := context.WithValue(r.Context(), contextKeyLogger, entry)
```

Comment:
entry isn't defined yet?   

You may need to setup 2 entries, 1 for the context logger and 1 for logging the response error.  Alternatively you could log the error and status code as the message instead of as fields and use the same entry for both